packages/leaderboard/src/upload.ts

```diff
@@ -0,0 +1,295 @@
+      }
+    }
+  )
+}
+
+async function updateNames(kit: ContractKit, rows: any[][], sheets: any) {
+  let accounts = await kit.contracts.getAccounts()
+  let data = []
+  for (let item of rows) {
+    if (!item[0]) data.push('')
+    else {
+      let name = await accounts.getName(item[0])
+      console.log('Name for', item[0], name)
+      data.push(name)
+    }
+  }
+  let req = {
+    spreadsheetId: LEADERBOARD_SHEET,
+    range: 'TGCSO!D3',
+    valueInputOption: 'USER_ENTERED',
+    requestBody: {
+      range: 'TGCSO!D3',
+      majorDimension: 'COLUMNS',
+      values: [data],
+    },
+  }
+  sheets.spreadsheets.values.update(req, (err: any, res: any) => {
+    console.log(res, err)
+  })
+}
+
+async function getBTUs(kit: ContractKit, address: string) {
+  const metadata = await getMetadata(kit, address)
+  const claimedAccounts = await getClaims(kit, address, metadata)
+
+  let sum = new BigNumber(0)
+  for (const address of claimedAccounts) {
+    try {
+      const balance = await kit.getTotalBalance(address)
+      sum = sum.plus(balance.total)
+    } catch (err) {
+      console.error('Error', err)
+    }
+  }
+  return sum.multipliedBy(new BigNumber('1e-18')).toString(10)
+}
+
+async function updateBTUs(kit: ContractKit, rows: any[][], sheets: any) {
+  let data = []
+  for (let item of rows) {
+    let address = item[0]
+    if (!address) data.push('')
+    else {
+      try {
+        let v = await getBTUs(kit, address)
+        console.log('BTU for', address, v)
+        data.push(v)
+      } catch (err) {
+        console.error('Cannot find BTU for', address, err)
+        data.push('')
+      }
+    }
+  }
+  let req = {
+    spreadsheetId: LEADERBOARD_SHEET,
+    range: 'TGCSO!E3',
+    valueInputOption: 'USER_ENTERED',
+    requestBody: {
+      range: 'TGCSO!E3',
+      majorDimension: 'COLUMNS',
+      values: [data],
+    },
+  }
+  sheets.spreadsheets.values.update(req, (err: any, res: any) => {
+    console.log(res, err)
+  })
+}
+
+async function updateAttestations(kit: ContractKit, rows: any[][], sheets: any) {
+  let data = []
+  let attestations = await kit._web3Contracts.getAttestations()
+  for (let item of rows) {
+    let address = item[0]
+    if (!address) data.push('')
+    else {
+      try {
+        let req = (
+          await attestations.getPastEvents('AttestationIssuerSelected', {
+            fromBlock: 0,
+            filter: { issuer: address },
+          })
+        ).length
+        let full = (
+          await attestations.getPastEvents('AttestationCompleted', {
+            fromBlock: 0,
+            filter: { issuer: address },
+          })
+        ).length
+        console.log('Attestations requested', req, 'fulfilled', full, 'by', address)
+        if (req == 0) data.push(0)
+        else data.push(full / req)
+      } catch (err) {
+        console.error('Cannot resolve attestations for', address, err)
+        data.push('')
+      }
+    }
+  }
+  let req = {
```

Comment:
seems like you could dedupe this with the above req and copy + spread only the `values: [data]` part